src/components/messageSpace/messageSpace.cy.tsx

```diff
@@ -77,16 +80,32 @@ describe('MessageSpace Component', () => {
         <MessageSpace
           messages={messages}
           supportedElements={supportedElements}
+          getProfileComponent={(message: ThreadableMessage) => {
+            if (message.sender.includes('Agent')) {
+              return <SmartToyIcon />
+            } else {
+              return <AccountCircleIcon />
+            }
+          }}
         />
       )
       const messageSpace = '[data-cy=message-space]'
 
       cy.get(messageSpace).should('exist')
 
-      messages.forEach((message) => {
+      messages.forEach((message, index) => {
         cy.get(messageSpace)
           .should('contain', message.sender)
           .and('contain', message.data.text)
+        if (message.sender === 'Agent') {
+          cy.get('svg')
+            .eq(index)
+            .should('have.attr', 'data-testid', `SmartToyIcon`)
```

Comment:
Maybe instead of relying on MUI's test attributes, we can directly add our own to the icons used above.

```suggestion
            .should('have.attr', 'data-cy', 'agent-icon')
```